src/snowflake/snowpark/modin/conftest.py

```diff
@@ -3,17 +3,20 @@
 # Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
 #
 
-import modin.pandas as pd  # pragma: no cover
-import numpy as np  # pragma: no cover
-import pytest  # pragma: no cover
+from snowflake.snowpark._internal.utils import COMPATIBLE_WITH_MODIN
 
-import snowflake.snowpark.modin.plugin  # pragma: no cover # noqa: F401
+# Guard against importing modin due to Python 3.8 issue.
+if COMPATIBLE_WITH_MODIN:
```

Comment:
I don't understand why this was added. Would CI normally run these tests on py3.8?

Reply:
Modin isn't compatible with Python 3.8 - We want to run AST tests with the package as is, given we plan to use (for now) python3.8 for the TCM. This is a fix to avoid the imports under python3.8 for Modin when testing. We do not run certain tests under 3.8 so that's why the issue was avoided/did not materialize :)